### PR TITLE
Resolves #604: Change "indexes need rebuilding" loggings to DEBUG and add a counter for it

### LINKED ISSUE #604: Change "indexes need rebuilding" loggings to DEBUG and add a counter for it
Currently, "indexes need rebuilding" is logged in INFO level. There can be too many logs if some indexes are never built. It would be better to change it to DEBUG while having a counter counting the number of indexes need to be built. The counter takes less space but provides enough information to warn developers to build indexes.

### PR DESCRIPTION
**Pull request by nblintao**

### CODE REVIEW

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ The Guava version has been updated to version 27. Users of the [shaded variants]
 * **Performance** Improvement 3 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Performance** Improvement 4 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Performance** Improvement 5 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
-* **Feature** Feature 1 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
+* **Feature** Added a counter counting indexes need to be rebuilt and changed related loggings to DEBUG [(Issue #604)](https://github.com/FoundationDB/fdb-record-layer/issues/604)
```
Comment (alecgrieser) [inline docs/ReleaseNotes.md]:
```suggestion
* **Feature** Added a counter counting indexes that need to be rebuilt and changed related logging to `DEBUG` [(Issue #604)](https://github.com/FoundationDB/fdb-record-layer/issues/604)
```

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimer.java [hunk 2]
```diff
@@ -452,6 +452,8 @@ public String title() {
         INVALID_SPLIT_SUFFIX("invalid split suffix", false),
         /** The number of record keys with an incorrect length found by {@link FDBRecordStore#repairRecordKeys(byte[], com.apple.foundationdb.record.ScanProperties)}. */
         INVALID_KEY_LENGTH("invalid record key", false),
+        /** The number of indexes need to be rebuilt in the record store. */
```
[the remark this answers is not in the document]
Comment (nblintao) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimer.java]: Done.

File: docs/ReleaseNotes.md (hunk 1 above)
Comment (nblintao) [reply docs/ReleaseNotes.md]: Done.

Comment (alecgrieser) [reply docs/ReleaseNotes.md]: Sorry for the nitpicking, but it looks like the "that" was added, but "loggings" -> "logging" and tick-marks around "DEBUG" were omitted.

Comment (nblintao) [reply docs/ReleaseNotes.md]: Thanks for pointing out! Fixed.

**Pull request merged**